tezos_interop/tezos_interop.re

```diff
@@ -506,24 +506,65 @@ module Run_contract = {
     };
   };
 };
-module Listen_transactions = {
-  open Tezos_micheline;
-  type michelson = Micheline.node(int, Michelson_v1_primitives.prim);
-  let michelson_of_yojson = json => {
-    // TODO: do this without serializing
-    let.ok json =
-      Yojson.Safe.to_string(json) |> Data_encoding.Json.from_string;
-    try(
-      Ok(
-        Micheline.root(
-          Data_encoding.Json.destruct(Pack.expr_encoding, json),
-        ),
-      )
-    ) {
-    | _ => Error("invalid json")
+
+type michelson =
+  Tezos_micheline.Micheline.node(int, Michelson_v1_primitives.prim);
+let michelson_of_yojson = json => {
+  // TODO: do this without serializing
+  let.ok json = Yojson.Safe.to_string(json) |> Data_encoding.Json.from_string;
+  try(
+    Ok(
+      Tezos_micheline.Micheline.root(
+        Data_encoding.Json.destruct(Pack.expr_encoding, json),
+      ),
+    )
+  ) {
+  | _ => Error("invalid json")
+  };
+};
+
+module Fetch_storage: {
+  let _run:
+    (~rpc_node: Uri.t, ~confirmation: int, ~contract_address: Address.t) =>
+    Lwt.t(result(michelson, string));
+} = {
+  [@deriving to_yojson]
+  type input = {
+    rpc_node: string,
+    confirmation: int,
+    contract_address: string,
+  };
+  [@deriving of_yojson]
+  type output = {
+    status: string,
+    storage: michelson,
+  };
```

Comment:
That's gonna fail to parse errors as they have the type

```reasonml
type output = {
  status: string,
  error: string
}
```

My suggestion, do a `| Success(michelson) | Error(string)` variant and decode it by hand